### PR TITLE
Trigger rhcos builds when rpms may have changed

### PR DESCRIPTION
**Pull request by jupierce**
When we update plashets (or when we are rebuilding images due to RHEL dependency updates), trigger RHCOS builds. If there are no changes, the build will quickly exit.

### CODE REVIEW

**Comment by sosiouxme** [comment]
/hold

there was a limitation in p8 capacity (IIRC) such that it doesn't handle more than two concurrent builds very well; the spread-out schedule we ended up with is intended to avoid that. any thoughts on how to prevent that being a problem when we no longer follow the schedule?

File: jobs/build/ocp4/Jenkinsfile [hunk 1]
```diff
@@ -156,14 +156,18 @@ node {
...
+                if ( params.ASSEMBLY == "stream" ) {
+                    // Since plashets may have been rebuilt, fire off sync for CI. This will transfer
+                    // RPMs out to mirror.openshift.com/enterprise so that they may be consumed through
+                    // CI rpm mirrors.
+                    build wait: false, propagate: false, job: '/scheduled-builds/sync-for-ci', parameters: [string(name: 'ONLY_FOR_VERSION', value: params.BUILD_VERSION)]
+
+                    // Also trigger rhcos builds for the release in order to absorb any changes from plashets or
+                    // RHEL which may have triggered our rebuild. If there are no changes to the RPMs, the build
+                    // should exit quickly. If there are changes, the hope is that by the time our images are done
+                    // building, RHCOS will be ready and build-sync will find consistent RPMs.
```
Comment (sosiouxme) [inline jobs/build/ocp4/Jenkinsfile:168]: that seems unlikely unless we're rebuilding a whole lot of non-RHCOS images. but in any case we will get new builds a great deal sooner than if we waited for the schedule, and `ocp4_scan` will incorporate them quickly.

**Comment by jupierce** [comment]
@sosiouxme I've added a lock that will let us control the number of synchronous executions by setting the resource count for a set of named locks: `lock("rhcos-build-capacity-${jobArch}")`

**Comment by sosiouxme** [comment]
/hold cancel
ppc64le has been de-scheduled so this is ready
/lgtm

it does occur to me, though, that this won't reliably get us past flakes. maybe as a refinement we should invoke this job in the `ocp4_scan` branch of logic where RHCOS has updates but isn't in sync yet.

**Pull request merged**